thunder/executors/torch_compile.py

```diff
@@ -75,23 +79,29 @@ def torch_interpreted_func(*args):
     # TODO: issue "Try using _transform_for_operator_executor_execution for
     # torch.compile executor"
     torch_trace = trace(inline_trace=False)(torch_interpreted_func, *sorted_unique_inputs)
-    compiled_func = torch.compile(torch_trace.python_callable())
-
+    # decorators are not added because dynamo graph breaks on our custom `no_autocast` decorator.
+    # so far it only supports their native `autocast` implementation
+    trace_callable = torch_trace.python_callable(include_decorators=False)
+    trace_callable = torch.autocast(device_type="cpu", enabled=False, cache_enabled=False)(trace_callable)
+    trace_callable = torch.autocast(device_type="cuda", enabled=False, cache_enabled=False)(trace_callable)
```

Comment:
I wonder if we could replace no_autocast by just these two applications of torch.autocast in a function 
https://github.com/Lightning-AI/lightning-thunder/blob/71fc8cda4e42b818372b6e9f1c99c9cc3a5c2e38/thunder/executors/torchex.py#L139

so
```
def no_autocast(fn):
    fn = torch.autocast(device_type="cpu", enabled=False, cache_enabled=False)(fn)
    fn = torch.autocast(device_type="cuda", enabled=False, cache_enabled=False)(fn)
    return fn
```

and then not having to special case anythinghere.

Reply:
Ah that's a good suggestion. Trying in https://github.com/Lightning-AI/lightning-thunder/pull/169/commits/d5614568fe833fd2ddcee5cffa39867668e4cd58.

Reply:
Actually, I need to partially revert this because even though I increased the cache size limit, this still causes extreme recompilation:

```python
V0422 14:59:42.353000 139978690109440 torch/_dynamo/guards.py:2122] [__recompiles] Recompiling function inner in /home/carlos/nightly-env/lib/python3.10/site-packages/torch/_dynamo/external_utils.py:34
V0422 14:59:42.353000 139978690109440 torch/_dynamo/guards.py:2122] [__recompiles]     triggered by the following guard failure(s):
V0422 14:59:42.353000 139978690109440 torch/_dynamo/guards.py:2122] [__recompiles]     - ___check_obj_id(L['fn'].__closure__[1].cell_contents.__closure__[0].cell_contents, 139974325237696)
V0422 14:59:42.353000 139978690109440 torch/_dynamo/guards.py:2122] [__recompiles]     - ___check_obj_id(L['fn'].__closure__[1].cell_contents.__closure__[0].cell_contents, 139974168693152)
V0422 14:59:42.353000 139978690109440 torch/_dynamo/guards.py:2122] [__recompiles]     - ___check_obj_id(L['fn'].__closure__[1].cell_contents.__closure__[0].cell_contents, 139974167264512)
V0422 14:59:42.353000 139978690109440 torch/_dynamo/guards.py:2122] [__recompiles]     - ___check_obj_id(L['fn'].__closure__[1].cell_contents.__closure__[0].cell_contents, 139974167080400)
V0422 14:59:42.353000 139978690109440 torch/_dynamo/guards.py:2122] [__recompiles]     - ___check_obj_id(L['fn'].__closure__[1].cell_contents.__closure__[0].cell_contents, 139974706455760)
V0422 14:59:42.353000 139978690109440 torch/_dynamo/guards.py:2122] [__recompiles]     - ___check_obj_id(L['fn'].__closure__[1].cell_contents.__closure__[0].cell_contents, 139974169152528)
V0422 14:59:42.353000 139978690109440 torch/_dynamo/guards.py:2122] [__recompiles]     - ___check_obj_id(L['fn'].__closure__[1].cell_contents.__closure__[0].cell_contents, 139974325238272)
```

So these context managers should definitely be left outside of the `torch.compile` region